integrationTests/chainSimulator/staking/delegation_test.go

```diff
@@ -750,6 +764,358 @@ func testChainSimulatorCreateNewDelegationContract(t *testing.T, cs chainSimulat
+	require.Nil(t, err)
+	require.NotNil(t, delegatorBTx2)
+	assert.Equal(t, delegatorBTx2.SmartContractResults[0].ReturnMessage, "total delegation cap reached")
+
+	// check the tx failed
+
+	output, err = executeQuery(cs, core.MetachainShardId, delegationContractAddress, "getTotalActiveStake", nil)
+	require.Nil(t, err)
+	require.Equal(t, expectedTotalStaked, big.NewInt(0).SetBytes(output.ReturnData[0]))
+	require.Zero(t, expectedTopUp.Cmp(getBLSTopUpValue(t, metachainNode, delegationContractAddress)))
+
+	output, err = executeQuery(cs, core.MetachainShardId, delegationContractAddress, "getUserActiveStake", [][]byte{delegatorBBytes})
+	require.Nil(t, err)
+	require.Zero(t, len(output.ReturnData))
+	require.Equal(t, "view function works only for existing delegators", output.ReturnMessage)
+
+	delegateValue = delegateValue.Sub(delegateValue, oneEGLD) // 500 EGLD
+	tx3delegatorB := generateTransaction(delegatorBBytes, 2, delegationContractAddress, delegateValue, "delegate", gasLimitForDelegate)
+	delegatorBTx3, err := cs.SendTxAndGenerateBlockTilTxIsExecuted(tx3delegatorB, maxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, delegatorBTx3)
+
+	expectedTopUp = expectedTopUp.Add(expectedTopUp, delegateValue)
+	expectedTotalStaked = expectedTotalStaked.Add(expectedTotalStaked, delegateValue)
+	output, err = executeQuery(cs, core.MetachainShardId, delegationContractAddress, "getTotalActiveStake", nil)
+	require.Nil(t, err)
+	require.Equal(t, expectedTotalStaked, big.NewInt(0).SetBytes(output.ReturnData[0]))
+	require.Equal(t, expectedTopUp, getBLSTopUpValue(t, metachainNode, delegationContractAddress))
+
+	output, err = executeQuery(cs, core.MetachainShardId, delegationContractAddress, "getUserActiveStake", [][]byte{delegatorBBytes})
+	require.Nil(t, err)
+	require.Equal(t, delegateValue, big.NewInt(0).SetBytes(output.ReturnData[0]))
+
+	delegateValue = big.NewInt(0).Mul(oneEGLD, big.NewInt(20)) // 20 EGLD
+	txDelegate3 := generateTransaction(delegatorCBytes, 0, delegationContractAddress, delegateValue, "delegate", gasLimitForDelegate)
+	delegatorCTx1, err := cs.SendTxAndGenerateBlockTilTxIsExecuted(txDelegate3, maxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, delegatorCTx1)
+	assert.Equal(t, delegatorBTx2.SmartContractResults[0].ReturnMessage, "total delegation cap reached")
+
+	output, err = executeQuery(cs, core.MetachainShardId, delegationContractAddress, "getTotalActiveStake", nil)
+	require.Nil(t, err)
+	require.Equal(t, expectedTotalStaked, big.NewInt(0).SetBytes(output.ReturnData[0]))
+	require.Equal(t, expectedTopUp, getBLSTopUpValue(t, metachainNode, delegationContractAddress))
+
+	output, err = executeQuery(cs, core.MetachainShardId, delegationContractAddress, "getUserActiveStake", [][]byte{delegatorCBytes})
+	require.Nil(t, err)
+	require.Zero(t, len(output.ReturnData))
+	require.Equal(t, "view function works only for existing delegators", output.ReturnMessage)
```

Comment:
The last txs, checks and so forth seem a bit duplicated.
Is there any way to create a func?

Reply:
there will be a refactor follow-up PR to remove duplicates